job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/DatabasePoller.java

```diff
@@ -0,0 +1,212 @@
+ */
+package com.hpe.caf.services.job.scheduled.executor;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hpe.caf.api.Codec;
+import com.hpe.caf.util.ModuleLoader;
+import com.hpe.caf.util.ModuleLoaderException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.StandardCharsets;
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.text.DateFormat;
+import java.text.MessageFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * This class is used to poll the Job Service database for jobs that can now run.
+ */
+public class DatabasePoller
+{
+    private static final Logger LOG = LoggerFactory.getLogger(DatabasePoller.class);
+
+    private static final String JDBC_POSTGRESQL_PREFIX = "jdbc:postgresql:";
+    private static final String JDBC_DRIVER = "org.postgresql.Driver";
+
+    public static void pollDatabaseForJobsToRun() {
+        try {
+            //  Poll database for prerequisite jobs that are now available to be run.
+            LOG.info("Polling Job Service database for jobs to run ...");
+            List<JobTaskData> jobsToRun = getDependentJobsToRun();
+
+            //  Determine if there are any jobs to run.
+            if (jobsToRun.size() > 0) {
+                //  Load serialization class.
+                LOG.debug("Loading serialization class ...");
+                Codec codec = ModuleLoader.getService(Codec.class);
+
+                //  For each job to run, submit message to the rabbitMQ queue for further processing.
+                for (com.hpe.caf.services.job.scheduled.executor.JobTaskData jtd : jobsToRun) {
+                    LOG.info(MessageFormat.format("Processing job id {0} ...", jtd.getJobId()));
+
+                    WorkerAction workerAction = new WorkerAction();
+                    workerAction.setTaskClassifier(jtd.getTaskClassifier());
+                    workerAction.setTaskApiVersion(jtd.getTaskApiVersion());
+                    if (jtd.getTaskData() != null) {
+                        workerAction.setTaskData(new String(jtd.getTaskData(), StandardCharsets.UTF_8));
+                    }
+                    workerAction.setTaskPipe(jtd.getTaskPipe());
+                    workerAction.setTargetPipe(jtd.getTargetPipe());
+
+                    //  Get database helper instance.
+                    try {
+                        QueueServices queueServices = QueueServicesFactory.create(jtd.getTaskPipe(),codec);
+                        LOG.debug(MessageFormat.format("Sending task data to the target queue {0} ...", workerAction.toString()));
+                        queueServices.sendMessage(jtd.getJobId(), workerAction);
+                        queueServices.close();
```

Comment:
Put the `close()` in a finally block as we need to ensure it is done even if there is an error.